### PR TITLE
fix: Install pip and pip-tools in upgrade script

### PR DESCRIPTION
**Pull request by Jawayria**
Updated the upgrade target script to check the compatibility of upgraded pip and pip-tools versions. For reference, look at this https://github.com/openedx/edx-repo-health/pull/271 for new check added in edx-repo-health.

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

**Comment by UsamaSadiq** [comment]
Need approval from the owning team as well.

File: requirements/common_constraints.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# A central location for most common version constraints
```
Comment (feanil) [inline requirements/common_constraints.txt]: @Jawayria should this file be a copy or should it just be linking to the central constraints file?

**Review by feanil — COMMENTED** [review]
One question, it seems like there shouldn't be a `common_constraints.txt` file in each repo since that would defeat the purpose of the common file. The comment in that file also suggests this.

File: requirements/common_constraints.txt (hunk 1 above)
Comment (Jawayria) [reply requirements/common_constraints.txt]: We mostly do this when we have to override some constraint but it isn't needed here so updated the PR

**Review by feanil — APPROVED** [review]
Looks like you had forgotten to delete the old requirements.txt file, I just added a commit for that onto this PR so now it's good to go.

**Comment by Jawayria** [comment]
> Looks like you had forgotten to delete the old requirements.txt file, I just added a commit for that onto this PR so now it's good to go.

Oh yes, Thanks for the commit

**Pull request merged**